core/src/main/java/org/apache/calcite/sql/SqlDialect.java

```diff
@@ -772,16 +773,26 @@ public boolean supportsDataType(RelDataType type) {
     return true;
   }
 
- /** Returns SqlNode for type in "cast(column as type)", which might be
+  public boolean allowTypeWithUnspecifiedPrecision(RelDataType type) {
+    return true;
+  }
```

Comment:
Can we give a document to this method ? `allowTypeWithUnspecifiedPrecision` -> `allowsTypeWithUnspecifiedPrecision`

Reply:
The document of this method has been added.